api/metric/api.go

```diff
@@ -72,6 +74,74 @@ func (m Measurement) Number() core.Number {
 	return m.number
 }
 
+// Kind describes the kind of instrument.
+type Kind int8
+
+const (
+	// MeasureKind indicates a Measure instrument.
+	MeasureKind Kind = iota
+	// ObserverKind indicates an Observer instrument.
+	ObserverKind
+	// CounterKind indicates a Counter instrument.
+	CounterKind
+)
+
+// Descriptor contains all the settings that describe an instrument,
+// including its name, metric kind, number kind, and the configurable
+// options.
+type Descriptor struct {
```

Comment:
So the prodigal son returns. ;)

Reply:
Maybe this should be called a `Definition` here, where it is returned as a struct value. In the export package, a pointer is returned allowing export pipelines to use the pointer value as a map key. The export type could be called `Descriptor`. I don't think the user cares too much, maybe there will be some reason a user wants this information, but as an implementor I think this makes sense.

Reply:
> Maybe this should be called a Definition here, where it is returned as a struct value. In the export package, a pointer is returned allowing export pipelines to use the pointer value as a map key. The export type could be called Descriptor. 

This is an interesting idea! I think it is probably better suited for a separate PR, but I like the concept of using a definition to define the instrument and a descriptor in the SDK to describe. That and the customized representation in either sounds like a usability benefit for implementors, SDK writers, and export writers.

Reply:
Maybe this can be done in #552?

Reply:
😅